src/transformers/trainer_tf.py

```diff
@@ -0,0 +1,490 @@
+# coding=utf-8
+""" Trainer class."""
+
+import os
+import logging
+from collections import OrderedDict
+import math
+import itertools
+from abc import ABC, abstractmethod
+
+import numpy as np
+
+from sklearn.metrics import classification_report
+
+import tensorflow as tf
+from .optimization_tf import WarmUp, AdamWeightDecay
+from .losses_tf import QALoss
+from .configuration_auto import AutoConfig
+from .tokenization_auto import AutoTokenizer
+from .data.processors.task_processors import DataProcessorForSequenceClassification
+from .modeling_tf_auto import TFAutoModelForSequenceClassification
+
+
+logger = logging.getLogger(__name__)
+
+
+class TFTrainer(ABC):
+    def __init__(self, **kwargs):
+        """
+        The list of keys in kwargs here should be generic to all the possible models/architectures
+        and not specific to such or such dataset/task.
+        """
+        self.pretrained_model_name_or_path = kwargs.pop("pretrained_model_name_or_path", None)
+        self.optimizer_name = kwargs.pop("optimizer_name", None)
+        self.warmup_steps = kwargs.pop("warmup_steps", None)
+        self.decay_steps = kwargs.pop("decay_steps", None)
+        self.learning_rate = kwargs.pop("learning_rate", None)
+        self.adam_epsilon = kwargs.pop("adam_epsilon", 1e-08)
+        self.loss_name = kwargs.pop("loss_name", None)
+        self.train_batch_size = kwargs.pop("train_batch_size", None)
+        self.eval_batch_size = kwargs.pop("eval_batch_size", None)
+        self.distributed = kwargs.pop("distributed", None)
+        self.epochs = kwargs.pop("epochs", None)
+        self.max_grad_norm = kwargs.pop("max_grad_norm", 1.0)
+        self.metric_name = kwargs.pop("metric_name", None)
+        self.max_len = kwargs.pop("max_len", None)
+        self.task = kwargs.pop("task", None)
+        self.datasets = {}
+        self.processor = None
```

Comment:
@jplu we try to use as many type hints as we can, especially since we've upgraded to Py36+

Here for instance it would be super helpful to know what type is the `processor`

Reply:
Ok perfect I will use the 3.6 typing. It will be added in the next commit 😊